### DISCUSSION TITLE
Dbeast monitor v0.7.1 Released!

### DISCUSSION THREAD

**Discussion by dbeast-co** [dbeast-co/dbeast-monitor#70] [Announcements]
Dbeast monitor v0.7.1 Released!

Full Changelog: https://github.com/dbeast-co/dbeast-monitor/wiki/ReleaseNotes
Installation Instructions: https://github.com/dbeast-co/dbeast-monitor/wiki/Installation-Instructions

<hr /><em>This discussion was created from the release <a href='https://github.com/dbeast-co/dbeast-monitor/releases/tag/DBeastMonitor-V-0.7.1'>Dbeast monitor v0.7.1 Released!</a>.</em>

**Comment by latsygin** [comment]
I have a problem.
After configuring Grafana, I see only one plugin in the interface: Dbeast Monitor. Why is there no Dbeast New cluster panel plugin?
P.S.: Grafana version - 9.5.3

**Reply by dbeast-co** [reply]
Hi,
Thank you for your response.
It looks like there is a different Grafana behavior in Linux and Windows... I'll test this issue.
Anyway, you can still install and use the application. The panel is present in the application, even if it doesn't appear in the plugins list.

**Reply by latsygin** [reply]
Please tell me how I can use the application if I cannot add a cluster (there is no corresponding section in the side menu - i.e. DBeast Monitor, see the screenshot)?
![Screenshot_2024_0624_121721](https://github.com/dbeast-co/dbeast-monitor/assets/32664599/b6b2a091-9567-41f8-9492-4bc92003144a)
P.S.: ...and yes, the operating system is Ubuntu 20.04 aarch64, but Grafana is running in docker :)

**Reply by dbeast-co** [reply]
When you press the "Add new cluster" do you see the cluster panel?
Like:
![image](https://github.com/dbeast-co/dbeast-monitor/assets/92628985/8ebff627-f225-41b6-a97e-7da77ab32292)

**Reply by latsygin** [reply]
You didn't understand - I don't have the "Add new cluster" button at all :(
![IMG_20240624_145423](https://github.com/dbeast-co/dbeast-monitor/assets/32664599/a82d956e-3b47-4c57-b054-1431a246dfe9)

**Reply by dbeast-co** [reply]
Maybe you don't enable the plugin? 
You are supposed to see the button with "Disable"
![image](https://github.com/dbeast-co/dbeast-monitor/assets/92628985/b9247c18-b765-42ca-a73f-4461201cb803)

**Reply by latsygin** [reply]
By the way! This is another interesting point that differs from the instructions - I have no tabs in the plugin window except for "Overview", and accordingly there are no "Enable/Disable" buttons. I don't understand what's going on, although I'm doing everything step by step...

**Reply by dbeast-co** [reply]
It's very strange. I tested this version today in your Grafana version, and there is no problem with it...
I'm not familiar with Docker, but if you want, we can schedule a meeting and check your installation

**Reply by latsygin** [reply]
Yes, it's strange. It is also strange that according to the instructions I have to unpack the archive (for my operating system) into a folder with plugins. But I don't find any archive in the repository, only the "dbeast-monitoring-for-elastic-stack-app" directory, which I use... Is this correct, or should there still be an archive for my operating system?
![Screenshot_2024_0624_161713](https://github.com/dbeast-co/dbeast-monitor/assets/32664599/fd79b796-7723-4e6f-a208-f15340e2c8fe)

**Reply by dbeast-co** [reply]
The dbeast-monitor-0.7.1-Linux.zip file that you are supposed to download from the release page contains dbeast-dbeastmonitor-app-0.7.1.tar.gz. This is the standard tar.gz archive for Linux. Unpack it into the plugin folder. You'll see the dbeast-dbeastmonitor-app folder, which contains the application

**Reply by latsygin** [reply]
Yes, I did everything wrong (didn't use the archive from the release page and instead used the directory in the repository...) and because of this there were these strange things... Thanks for the help! 

Now I enabled the plugin, but encountered an error when adding a cluster, when I clicked on the "Test" button... I am attaching a screenshot.
![image](https://github.com/dbeast-co/dbeast-monitor/assets/32664599/0ccd439f-878b-4238-81e1-76d31b73ff0c)

**Reply by dbeast-co** [reply]
Can you please share the Grafana logs?

**Reply by latsygin** [reply]
```
logger=context userId=1 orgId=1 uname=admin t=2024-06-24T18:43:25.323791139Z level=warn msg="Plugin unavailable" error="plugin unavailable" traceID=
logger=context userId=1 orgId=1 uname=admin t=2024-06-24T18:43:25.32406093Z level=error msg="Request Completed" method=POST path=/api/plugins/dbeast-dbeastmonitor-app/resources/test_cluster status=503 remote_addr=192.168.65.1 time_ms=8 duration=8.70875ms size=88 referer="http://0.0.0.0:3000/d/add-new-cluster-dashboard/add-new-cluster-dashboard?orgId=1" handler=/api/plugins/:pluginId/resources/*
```
I don’t understand what kind of address 192.168.65.1 is...

**Reply by dbeast-co** [reply]
It's really strange. The application is supposed to connect to the internal backend. It looks like some Docker image misconfiguration. I'll try to reproduce this, but it can take time...

**Reply by latsygin** [reply]
I installed grafana on the server (IP is 10.0.0.60) - the problem persisted. Apparently the point is that the request is made taking into account the Linux routing table.
Here are the VM network interface settings
```
> ip a 
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 00:0c:29:2d:fa:ca brd ff:ff:ff:ff:ff:ff
    inet 172.16.231.129/24 brd 172.16.231.255 scope global dynamic eth0
       valid_lft 1391sec preferred_lft 1391sec
    inet6 fe80::20c:29ff:fe2d:faca/64 scope link 
       valid_lft forever preferred_lft forever
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 00:0c:29:2d:fa:d4 brd ff:ff:ff:ff:ff:ff
    inet 10.0.0.60/24 brd 10.0.0.255 scope global eth1
       valid_lft forever preferred_lft forever
    inet6 fe80::20c:29ff:fe2d:fad4/64 scope link 
       valid_lft forever preferred_lft forever
```
Here is the Grafana log:
```
logger=context userId=1 orgId=1 uname=admin t=2024-06-26T10:09:24.639331922Z level=warn msg="Plugin unavailable" error="plugin unavailable" traceID=
logger=context userId=1 orgId=1 uname=admin t=2024-06-26T10:09:24.639513087Z level=error msg="Request Completed" method=POST path=/api/plugins/dbeast-dbeastmonitor-app/resources/test_cluster status=503 remote_addr=10.0.0.1 time_ms=6 duration=6.343484ms size=88 referer="http://10.0.0.60:3000/d/add-new-cluster-dashboard/add-new-cluster-dashboard?orgId=1" handler=/api/plugins/:pluginId/resources/*
```
In the log we see the address 10.0.0.1, it corresponds to the routing settings for the eth1 interface:
```
> ip route
default via 172.16.231.2 dev eth0 proto dhcp src 172.16.231.129 metric 100 
10.0.0.0/24 dev eth1 proto kernel scope link src 10.0.0.60 
172.16.231.0/24 dev eth0 proto kernel scope link src 172.16.231.129 
172.16.231.2 dev eth0 proto dhcp scope link src 172.16.231.129 metric 100 
```
Question: how should I configure routing so that the plugin sends the request to where it needs to go? (to address 10.0.0.70, see screenshot above)

**Reply by dbeast-co** [reply]
Have you defined this address in the Grafana?

**Reply by latsygin** [reply]
yes, I specify the address 10.0.0.70 in Grafana (this can be seen in the screenshot above)

**Reply by dbeast-co** [reply]
But in the "ip a" results, your ip: 10.0.0.60 and not 70

**Reply by latsygin** [reply]
10.0.0.60 - Grafana
10.0.0.70 - ELK (elastic + logstash + kibana)

**Reply by dbeast-co** [reply]
It's extraordinary... you are supposed to see your cluster address here
BTW, I record the video with the application installation guide. He shows only the installation part, but maybe it help you...
I plan to record the "Add new cluster" part next week.
[https://www.youtube.com/watch?v=-i_9fsjk_iw](https://www.youtube.com/watch?v=-i_9fsjk_iw)
